### ISSUE TITLE
Add TLS support for all applicable sources

### ISSUE THREAD

**Issue by LucioFranco** [timberio/vector#1553]
Currently, none of our sources have the ability to configure TLS. To be consitient with our sinks we should provide TLS support to ensure vector can securely communicate on both the receiving end and the sending end.

## Proposal

This is a bit higher level but should have the same settings across all sources. Since, a lot of our sources are built from the `socket` source we should be able to apply a generic solution. This should also be done similarly to the generic tls implementation we have for sinks.

### Config

```toml
[source.in.tls]
enabled = true
crt_path = "..."
key_path = "..."
key_pass = "..."
```

Each sink that is based off of the tcp source can use these settings. The implementation should use a combination of `openssl` and `tokio-openssl`, we can also probably reuse configuration parsing from the sinks tls implementation. This should also be expandable to the http based sources since they work off of the same `accept(2)` syscall.

### kafka

Since, the kafka source doesn't need a tcp server but is just a tcp client, we should be able to reuse the same config options from the sink.

## Questions

- Do we want to support mTLS? This would include an extra `client_ca_cert` to validate against

**Comment by binarylogic** [comment]
@bruceg I'm assigning this to you since you worked on the TLS settings for the sinks. Let me know if you think this needs further spec'ing.

**Comment by bruceg** [comment]
The existing TLS support is largely based on `native_tls`. As per #1402, should I be basing any new support on the `openssl` crate directly, or continue to use `native_tls` and switch it over separately?

**Comment by binarylogic** [comment]
@LucioFranco should be able to answer that.

**Comment by LucioFranco** [comment]
@bruceg yeah, lets do as much via openssl/hyper-openssl for now. If things need to go in other PRs that is fine but our end goal should be to 100% be using openssl thus ensuring that we use the same version across all our sources/transforms/sinks.

**Comment by bruceg** [comment]
List of all sources and status:

Source | Mode | Status
-------|------|-------
docker | local | N/A
file | local | N/A
journald | local | N/A
kafka | connector | #1893 
kubernetes | local | N/A
logplex | acceptor | incomplete
prometheus | connector | done
socket | acceptor | #1892 
splunk_hec | acceptor | incomplete
statsd | acceptor | N/A (uses UDP)
stdin | local | N/A
syslog | acceptor | #1892 
vector | acceptor | #1892

**Comment by bruceg** [comment]
I believe with #1968 being merged to close #1904, this task is now also complete.

**Comment by LucioFranco** [comment]
Technically on that list docker already uses TLS, but otherwise I think we can close this!
